microhapulator/marker.py

```diff
@@ -47,27 +47,41 @@ def from_files(cls, csv_path, fasta_path=None):
     @staticmethod
     def parse_definitions_from_csv(path):
         table = pd.read_csv(path, sep=None, engine="python")
-        missing = set(["Marker", "Offset"]) - set(table.columns)
-        if len(missing) > 0:
-            missing = ", ".join(sorted(missing))
-            message = f"column(s) missing from marker definition file: {missing}"
-            raise ValueError(message)
+        MicrohapIndex.check_definition_columns(table)
         loci = defaultdict(Locus)
         for markerid, rowgroup in table.groupby("Marker"):
-            chrom = None
-            offchr = None
-            if "OffsetHg38" in rowgroup and not rowgroup.OffsetHg38.isnull().all():
-                offchr = sorted(rowgroup.OffsetHg38)
-            if "Chrom" in rowgroup:
-                if len(rowgroup.Chrom.unique()) > 1:
-                    chroms = ", ".join(sorted(rowgroup.Chrom.unique()))
-                    message = f"ambiguous chromosome definition for {markerid}: {chroms}"
-                    raise ValueError(message)
-                chrom = rowgroup.Chrom.iloc[0]
-            marker = MarkerDefinition(markerid, rowgroup.Offset, offsets_chrom=offchr, chrom=chrom)
+            offsets = rowgroup.Offset
+            chrom = MicrohapIndex.marker_chromosome(rowgroup)
+            chrom_offsets = MicrohapIndex.chromosome_offsets(rowgroup)
+            marker = MarkerDefinition(markerid, offsets, chrom_offsets, chrom)
```

Comment:
The refactoring here marginally increased the lines of code in exchange for some improved clarity.